src/colorizePrint.ts

```diff
@@ -0,0 +1,97 @@
+import * as vscode from 'vscode';
+import { ColorArray, colorTo, ColorType, roundColor } from './util/colorMap';
+
+/**
+ * Matching:
+ * ```
+ * new Color3( 0.1, 0.2, 0.3 );
+ * ----------  ---  ---  ---
+ *           --   --   --   --
+ * 1         2 3  4 5  6 7  8
+ * ```
+ */
+const matchColors: Record<ColorType, RegExp> = {
+	[ColorType.new]: /(new\s+Color3)(\(\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*\))/,
+	[ColorType.fromRGB]: /(Color3\s*\.fromRGB)(\s*\(\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*\))/,
+	[ColorType.fromHSV]: /(Color3\s*\.fromHSV)(\s*\(\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*,\s*)(\d*(?:\.\d*)?)(\s*\))/
+};
+
+const matchPrefix: Record<ColorType, string> = {
+	[ColorType.new]: 'new Color3',
+	[ColorType.fromRGB]: 'Color3.fromRGB',
+	[ColorType.fromHSV]: 'Color3.fromHSV',
+};
+
+function formatNumber(match: ColorType, a: number, b: number, c: number): vscode.Color {
+	if (match === ColorType.new) {
+		return {
+			alpha: 1,
+			red: a,
+			green: b,
+			blue: c
+		};
+	}
+
+	const [red, green, blue] = colorTo[match][ColorType.new](a, b, c);
+
+	return {
+		alpha: 1,
+		red,
+		green,
+		blue
+	};
+}
+
+function replaceMatch(match: RegExpMatchArray, group: string, a: number | string, b: number | string, c: number | string) {
+	return `${group}${match[2]}${a}${match[4]}${b}${match[6]}${c}${match[8]}`;
+}
+
+function extractTriColor(match: RegExpMatchArray): ColorArray {
+	return [Number(match[3]), Number(match[5]), Number(match[7])];
+}
+
+export function makeColorProvider() {
+	return vscode.languages.registerColorProvider("typescript", {
+		provideColorPresentations: (color, context, token) => {
+			const text = context.document.getText(context.range);
+
+			const match = Object.values(ColorType).find(match => text.includes(match));
+			if (!match) throw new Error('Color type specified was not found!');
+
+			const regexMatch = text.match(matchColors[match])!;
+
+			const matches: vscode.ProviderResult<Array<vscode.ColorPresentation>> = Object.entries(ColorType).map(([matchTypeRaw]): vscode.ColorPresentation => {
+				const matchType = matchTypeRaw as ColorType;
```

Comment:
Is there a reason for the second variable and type cast here? Couldn't you just set the parameter as `[, matchType]` to grab the ColorType?

Reply:
No there isn't, I'll remove that

Reply:
On second thought, there isn't a need for entries anyways, I'll replace that with values